### PR TITLE
Add option for custom_key_filters

### PR DESCRIPTION
**Pull request by mikker**
Closes #127. Thank you @Bugagazavr!

### CODE REVIEW

**Comment by mikker** [comment]
Also merged [this](https://github.com/elastic/apm-agent-ruby/pull/125) which adds a bit of confusion to the diff (and will probably give me a headache when I merge it 🙄)

File: lib/elastic_apm/filters/secrets_filter.rb [hunk 1]
```diff
@@ -19,8 +19,11 @@ class SecretsFilter
         /^\d{4}[- ]?\d{4}[- ]?\d{4}[- ]?\d{4}$/ # (probably) credit card number
       ].freeze
 
+      attr_reader :config
+
       def initialize(config)
         @config = config
+        @key_filters = KEY_FILTERS + config.custom_header_filters
```
Comment (Bugagazavr) [inline lib/elastic_apm/filters/secrets_filter.rb]: So it called `custom_header_filters`, but it worked for all type of keys. Thats not problem if you want to have separated configuration for headers, params, values in future.

File: lib/elastic_apm/filters/secrets_filter.rb [hunk 2]
```diff
@@ -19,8 +19,11 @@ class SecretsFilter
         /^\d{4}[- ]?\d{4}[- ]?\d{4}[- ]?\d{4}$/ # (probably) credit card number
       ].freeze
 
+      attr_reader :config
```
Comment (Bugagazavr) [inline lib/elastic_apm/filters/secrets_filter.rb]: I guess you can drop it.

File: lib/elastic_apm/filters/secrets_filter.rb (hunk 1 above)
Comment (mikker) [reply lib/elastic_apm/filters/secrets_filter.rb]: You are right. I'll consider moving back.

File: lib/elastic_apm/filters/secrets_filter.rb (hunk 2 above)
Comment (mikker) [reply lib/elastic_apm/filters/secrets_filter.rb]: Nice catch! 👍🏼

File: docs/configuration.asciidoc [hunk 3]
```diff
@@ -15,20 +13,334 @@ server_url: 'http://localhost:8200'
...
+==== `custom_key_filters`
+[options="header"]
+|============
+| Environment                      | `Config` key         | Default | Example
+| `ELASTIC_APM_CUSTOM_KEY_FILTERS` | `custom_key_filters` | `[]`    | `['MyAuthHeader']`
+|============
+
+Elastic APM strips
+https://github.com/elastic/apm-agent-ruby/blob/1.x/lib/elastic_apm/filters/secrets_filter.rb[
+what looks like confidential information] from the request/response headers.
```
Comment (Bugagazavr) [inline docs/configuration.asciidoc]: > from the request/response headers, params

File: docs/configuration.asciidoc [hunk 4]
```diff
@@ -15,20 +13,334 @@ server_url: 'http://localhost:8200'
...
+Use this option to add your own custom header keys to the list of filtered keys.
```
Comment (Bugagazavr) [inline docs/configuration.asciidoc]: > Use this option to add your own custom header, param keys to the list of filtered keys.

File: docs/configuration.asciidoc (hunk 3 above)
Comment (mikker) [reply docs/configuration.asciidoc]: Until we actually do strip them from the params, I want to keep it to the headers.

**Comment by mikker** [comment]
💙💚💛💜❤️

**Pull request merged**